pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -393,6 +412,7 @@ func (d *DeploymentController) render(templateName string, mi TemplateInput) ([]
 		MeshConfig:  cfg.MeshConfig,
 		Values:      cfg.Values.Map(),
 	}
+	setInputValue(input, "global.multiCluster.clusterName", d.clusterID)
```

Comment:
can you move this into `d.injectConfig()` looks not appropriate to handle it here